eng/Directory.Build.Data.props

```diff
@@ -47,6 +47,8 @@
     <EnableFxCopAnalyzers Condition="'$(IsShippingClientLibrary)' == 'true'">true</EnableFxCopAnalyzers>
     <EnableStyleCopAnalyzers Condition="'$(EnableStyleCopAnalyzers)' == '' and '$(IsClientLibrary)' == 'true'">true</EnableStyleCopAnalyzers>
     <GenerateAPIListing Condition="'$(IsShippingClientLibrary)' == 'true'">true</GenerateAPIListing>
+    <UpdateSourceOnBuild Condition="'$(UpdateSourceOnBuild)' == ''">$(AZURE_DEV_UPDATESOURCESONBUILD)</UpdateSourceOnBuild>
```

Comment:
Could we add a quick description of what `AZURE_DEV_UPDATESOURCESONBUILD` does and valid values for it in Contributing.md or somewhere similar?  I don't think anyone's ever going to find it otherwise.

Reply:
Added with this PR https://github.com/Azure/azure-sdk-for-net/pull/13721